### PR TITLE
add todos list

### PR DESCRIPTION
**Pull request by Misha408**
[DEMO LINK](https://misha408.github.io/react_static-list-of-todos-js/)

### CODE REVIEW

File: src/components/TodoInfo/TodoInfo.jsx [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+export const TodoInfo = ({ todo }) => (
+  <article className={cn('TodoInfo', {
+    'TodoInfo--completed': todo.completed === true,
+  })}
+  >
+    <h2 className="TodoInfo__title">
+      {todo.title}
+    </h2>
+
+    <UserInfo user={todo.user} />
```
Comment (vlad-demchuk) [inline src/components/TodoInfo/TodoInfo.jsx:13]: destructure todo to avoid repetition

File: src/components/UserInfo/UserInfo.jsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
-// Add the required props
-export const UserInfo = () => (
-  <>UserInfo markup</>
+export const UserInfo = ({ user }) => (
+  <a className="UserInfo" href={`mailto:${user.email}`}>
+    {user.name}
```
Comment (vlad-demchuk) [inline src/components/UserInfo/UserInfo.jsx:3]: the same

**Review by vlad-demchuk — CHANGES_REQUESTED** [review]

File: src/components/TodoInfo/TodoInfo.jsx [hunk 3]
```diff
@@ -1,4 +1,19 @@
-// Add the required props
-export const TodoInfo = () => (
-  <>TodoInfo markup</>
-);
+import cn from 'classnames';
+import { UserInfo } from '../UserInfo';
+
+export const TodoInfo = ({ todo }) => {
+  const todos = { ...todo };
```
Comment (Anastasiia-Svintsova) [inline src/components/TodoInfo/TodoInfo.jsx:5]: You don't need to create a copy of todo

File: src/components/TodoInfo/TodoInfo.jsx [hunk 4]
```diff
@@ -1,4 +1,19 @@
...
+
+  return (
+    <article className={cn('TodoInfo', {
+      'TodoInfo--completed': todos.completed === true,
+    })}
+    >
+      <h2 className="TodoInfo__title">
+        {todos.title}
```
Comment (Anastasiia-Svintsova) [inline src/components/TodoInfo/TodoInfo.jsx:13]: Use destructuring for objects for better code readability

File: src/components/UserInfo/UserInfo.jsx [hunk 5]
```diff
@@ -1,4 +1,9 @@
-// Add the required props
-export const UserInfo = () => (
-  <>UserInfo markup</>
-);
+export const UserInfo = ({ user }) => {
+  const users = { ...user };
```
Comment (Anastasiia-Svintsova) [inline src/components/UserInfo/UserInfo.jsx:2]: Same. Why do you need a copy here?

File: src/components/UserInfo/UserInfo.jsx [hunk 6]
```diff
@@ -1,4 +1,9 @@
...
+
+  return (
+    <a className="UserInfo" href={`mailto:${users.email}`}>
+      {users.name}
```
Comment (Anastasiia-Svintsova) [inline src/components/UserInfo/UserInfo.jsx:6]: The same about the destructuring

**Review by Anastasiia-Svintsova — CHANGES_REQUESTED** [review]

File: src/components/TodoInfo/TodoInfo.jsx [hunk 7]
```diff
@@ -1,4 +1,15 @@
...
+export const TodoInfo = ({ todo }) => (
+  <article className={cn('TodoInfo', {
+    'TodoInfo--completed': todo.completed === true,
+  })}
+  >
+    <h2 className="TodoInfo__title">
+      { todo.title }
+    </h2>
+
+    <UserInfo user={todo.user} />
```
Comment (TarasHoliuk) [inline src/components/TodoInfo/TodoInfo.jsx:13]: the `getUserById` function may return `null`, i.e. `todo.user` might be null theoretically
you should handle this case (do not render the UserInfo component at all in this case) - use conditional rendering

File: src/components/TodoInfo/TodoInfo.jsx [hunk 8]
```diff
@@ -1,4 +1,15 @@
...
+import cn from 'classnames';
+import { UserInfo } from '../UserInfo';
+
+export const TodoInfo = ({ todo }) => (
+  <article className={cn('TodoInfo', {
+    'TodoInfo--completed': todo.completed === true,
+  })}
+  >
+    <h2 className="TodoInfo__title">
+      { todo.title }
```
Comment (TarasHoliuk) [inline src/components/TodoInfo/TodoInfo.jsx:10]: take into account the previous comment about destructuring assignment (everywhere)
if you have questions - feel free to ask in the chat

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]

**Review by volodymyr-soltys97 — APPROVED** [review]
Nice work 👍

**Pull request closed without merging**